69 hidden website arabic i18 next translator

messam10 opened the pull request:
### Title: Add the translator text
-----
**Overview:**
This pull request Add the translator text to the site
-----
**Resolved issue:**
69 hidden website Arabic i18 next translator  #69
-----
**Notes to the reviewer:**
Please make sure the translator is so good or if there is any improvement
-----
**Screenshots/Recordings:**
![Screenshot (27)](https://user-images.githubusercontent.com/49818381/235807184-b2d5c92e-c3de-4f4a-a8c3-60c9107c25b8.png)

![Screenshot (28)](https://user-images.githubusercontent.com/49818381/235807211-f35a4c22-b00f-41eb-957e-2682836eb664.png)

![Screenshot (29)](https://user-images.githubusercontent.com/49818381/235807231-f0d90fd3-d87a-44f1-8ba7-1df357d38e26.png)

AbobakrMahdii committed:
refactor(aboutus): test to transulateing

messam10 committed:
fix: merge with develop branch

messam10 committed:
refactor(navbar): translator with Link buttons languages

messam10 committed:
fix(package): fs is available on the site

The error is because angular-cli does not support modules in nodes like "fs" and "path".

messam10 committed:
test(navbar): test text in another pages

messam10 committed:
feat(navbar): navbar component translation

messam10 committed:
style(navbar): activating a trend for Arabic and English languages

messam10 committed:
fix(navbar): remove dir i18next import

messam10 committed:
refactor(aboutus): translate a page about us

messam10 committed:
test(aboutus): run test for snapshot

messam10 committed:
fix(aboutus): remove text in describe

messam10 committed:
feat(ourteam): add translation functions to our team component

messam10 committed:
refactor(ourteam): translation of our team component

messam10 committed:
test(ourteam): run npm test and check files

messam10 committed:
feat(footer): add translation functions to footer component

messam10 committed:
refactor(footer): translation of the footer component

messam10 committed:
feat(donation): add translation functions to donation component

messam10 committed:
refactor(donation): translation of the donation component

messam10 committed:
feat(index): add translation functions to filter-buttons component

messam10 committed:
feat(index): add translation functions to filter-category component

messam10 committed:
feat(index): add translation functions to filter-price component

messam10 committed:
feat(productlist): add translation functions to product-list component

messam10 committed:
feat(productlist): add translation functions to product-data

messam10 committed:
test(productlist): test function translator

messam10 committed:
refactor(productlist): translation of the product-list / price-filter component

messam10 committed:
refactor(productlist): add arabic translator

messam10 committed:
refactor(index): translation of the product-list component

messam10 committed:
feat(signin): add translation functions to sign-in component

messam10 committed:
refactor(signin): translation of the sign-in component

messam10 committed:
feat(signup): add translation functions to sign-up component

messam10 committed:
refactor(signup): translation of the sign-up component

messam10 committed:
fix(singup): reset pages with component and link

messam10 committed:
fix: modify the link in the navbar and the footer

messam10 committed:
feat(404-page): add translation functions to 404 page with translator text

messam10 committed:
fix(404page): add translation functions to the page for navbar and footer

messam10 committed:
fix(signup): move files for page to component

messam10 committed:
test: run npm test with lint with prettier

messam10 committed:
style(navbar): modification of the design to suit both Arabic and English languages

messam10 committed:
style(footer): modification of the design to suit both Arabic and English languages

messam10 committed:
style(footer): improve text display

messam10 committed:
style(footer): edit widht size for texts

messam10 committed:
style(navbar): modify the display of the Language dialog

messam10 committed:
feat(userorder): add translation functions to the page

messam10 committed:
refactor(userorder): translate the user-order component

messam10 committed:
feat(userlistings): add translation functionality to the component

messam10 committed:
refactor(userlastings): translate the user-listings component

messam10 committed:
fix(userpage): modify the appearance of a tab with sidebar

messam10 committed:
feat(editprofile): add translation functionality to the edit-profile component

messam10 committed:
refactor(editprofile): translate the edit-profile component

messam10 committed:
refactor(editprofile): snapshot with add translator arabic

messam10 committed:
feat(sidebar): add translation functionality to the component sidebar

messam10 committed:
refactor(sidebar): translate the sidbar component

messam10 committed:
refactor(sidebar): add title for highlighter Style

messam10 committed:
refactor(sign in): translate the sign in component

messam10 committed:
fix: merge develop branch

messam10 committed:
fix: run test for all files

messam10 committed:
fix(about us): fix translator functions

messam10 committed:
fix(sign in): add translator for sign in component

messam10 committed:
fix(sign up): remove folder sign up form pages and install React-Toastify

messam10 committed:
fix: rename sign in & sign up

messam10 committed:
fix: update files for functions translator

messam10 committed:
fix: replace translator form react to next

messam10 committed:
style(navbar): edit languages show and hidden with rtl

messam10 committed:
style(navbar): modify the display of the Language dialog for responsively

messam10 committed:
style(navbar): modify logo by screen size and language

messam10 committed:
style(footer): edit text by screen size and language

messam10 committed:
test: run npm test for files

messam10 committed:
style(navbar): modify the logo by screen

messam10 committed:
style(footer): edit text and logo by screen

messam10 committed:
test(footer): npm run test

messam10 committed:
fix: merge with develop branch

messam10 committed:
feat(favproducts): add translation functionality to the component fraProduct

messam10 committed:
refactor(favproducts): translate the component

messam10 committed:
fix: run npm lint for files

messam10 committed:
fix: run npm prettier with lint

nabilashbat commented on package.json:
Can explain why you added these changes?

messam10 replied:
I ran into an fs issue where it appears not to exist...

And I found this solution
[Solution](https://bobbyhadz.com/blog/module-not-found-cant-resolve-fs)

AbobakrMahdii reviewed (approved):
Very very very very awesome work bro 🤤

finally the translating uploaded 😂

nabilashbat commented on src/components/Navbar/index.jsx:
```diff
@@ -70,20 +82,20 @@ export default function Navbar() {
               href='/donation'
               className='relative mx-4 capitalize text-sm hover:text-violet-800   transition-all duration-1000 before:absolute before:-bottom-2 before:left-0  before:h-[1px]  before:w-0 before:opacity-0 before:bg-violet-700 before:transition-all before:duration-500 hover:before:w-full hover:before:opacity-100'
```
Why not cleaning the code and extracting these classes into a variable and use this variable instead of repeating the classes here and down this one??

messam10 replied:
Well, it's better to create an issue to clean the code from all components.

AbobakrMahdii reviewed (changes requested):
Bro @messam10 there is a problem in the filters for both the favorited page and home page while language is arabic it never work also in the favorited page it making a crach that stops the hole website you need to fix it

messam10 committed:
fix(favproducts): change condition text to resolve filtering issue

solve for Application error: a client-side exception has occurred (see the browser console for more
information).

messam10 committed:
fix(favproducts): activate filtering in Arabic and English

messam10 committed:
fix(category): activate filtering in Arabic and English

messam10 committed:
style(product list): edit where the category is displayed by language

messam10 committed:
style(favproducts): edit where the category is displayed by language

messam10 committed:
test: run npm test

AbobakrMahdii reviewed (approved):
Good job

messam10 committed:
fix: merge with develop branch

AbobakrMahdii committed:
Merge branch 'develop' into 69-hidden-website-arabic-i18-next-translator

messam10 committed:
Merge branch '69-hidden-website-arabic-i18-next-translator' of github.com:202212-GIZ-YE-FEW/recoded_student-store_codeForce-cap into 69-hidden-website-arabic-i18-next-translator
